fluss-server/src/main/java/com/alibaba/fluss/server/log/LogTablet.java

```diff
@@ -425,7 +428,6 @@ public Optional<LogOffsetMetadata> maybeIncrementHighWatermark(
             if (oldHighWatermark.getMessageOffset() < newHighWatermark.getMessageOffset()
                     || (oldHighWatermark.getMessageOffset() == newHighWatermark.getMessageOffset()
                             && oldHighWatermark.onOlderSegment(newHighWatermark))) {
-                updateHighWatermarkMetadata(newHighWatermark);
```

Comment:
This makes the `maybeIncrementHighWatermark` very confusing and very error-prone. 

1. The `maybeIncrementHighWatermark` actually won't update the highwatermark underlying, which is inconsistent with the method name (even the Javadoc clarifies, but if a method name needs Javadoc to explain for it, the method is not a good design).
2. The watermark updates become non-atomic as it is in two different locks.